### DISCUSSION TITLE
Is there any documentation for releases?

### DISCUSSION THREAD

**Discussion by carolili** [k8ssandra/k8ssandra#656] [Q&A]
Hello, I can't seem to find any documentation for how/when releases are done. I was looking for the latest version (since bugfixes are added quite frequently, thanks for that work! 😃) and found the following:

- A [release tagged as v1.0.0](https://github.com/k8ssandra/k8ssandra/releases/tag/v1.0.0)
- A k8ssandra-next repo: https://helm.k8ssandra.io/next found [in an issue discussion](https://github.com/k8ssandra/k8ssandra/issues/599)
  - This repo contains minor releases `1.0.4` and `1.0.5` which do not seem to be tagged in this repo as well as `--devel` releases.
- The `main` branch of this repo.

I would have two questions:
- Is the `main` branch somewhat stable to try to deploy or is it expected that it can contain errors? E.g. for now I am seeing the following:
```
Helm install failed: template: k8ssandra/templates/stargate/stargate.yaml:22:3:
executing "k8ssandra/templates/stargate/stargate.yaml at <include "k8ssandra.labels" .>:
error calling include: template: k8ssandra/templates/_helpers.tpl:42:4:
executing "k8ssandra.labels" at <include "k8ssandra-common.labels" .>:
error calling include: template: no template "k8ssandra-common.labels" associated with template "gotpl"
```
- Is there a regular release schedule (there were some releases being tagged up until v1.0.0) or any documentation where users can see when a new release is planned? I found the [roadmap](https://github.com/orgs/k8ssandra/projects/6), but it's not clear if the Q1-Q4 correspond to releases.

Thanks for clarifying 😃

**Comment by jsanda** [comment]
Hi @carolili 

We have been working on the release/version processes. The intention was to have the two Helm repos as we currently do - _next_ and _stable_. Every commit to the ` main` branch would result in charts being updated in the _next_ repo. The _stable_ repo is updated when we do a project release. #599 complicates this a bit.

We do intend to have a stable `main`. A lot of work has been done to add and improve test automation in CI, and there is more to come. This will hopefully help us maintain a stable `main`.

We would like to get on a monthly release cadence, but it might take a couple releases before that happens as we sort out processes for releasing, versioning charts, etc.

I do not think that the Q1-Q4 on the roadmap correspond directly to releases but instead to the timeframe where would like to see these things done.

**Comment by jsanda** [comment]
I should have also mentioned that we just released 1.1 at the end of last week. See [CHANGELOG-1.1.md](https://github.com/k8ssandra/k8ssandra/blob/main/CHANGELOG-1.1.md) for what is included.

Notice that there is also a [CHANGELOG-1.2.md](https://github.com/k8ssandra/k8ssandra/blob/main/CHANGELOG-1.2.md) where changes will go for the next planned release.

**Comment by jdonenine** [comment]
Thanks for the questions @carolili !

@jsanda 's answers were great and we're definitely working towards a more cadenced planning/release process.  As John mentions, the current roadmap isn't a direct alignment to release, and more a bigger bucket prioritization.  But, a more clear alignment to releases is something we're talking about adding to the roadmap project.

I'm curious, what sort of information would you find most useful on the release front?  Are you just interested in at what cadence to expect new releases? Looking to understand the future content?  Looking to influence the content?  All of the above?

Would be awesome to hear a bit more about how you're using k8ssandra and what you're looking for in the future from the project.

Also @bradfordcp it sounds like we perhaps need some more docs on the site about this topic:  Cadence, Where to find the roadmap, What the release streams (stable/next) are and how they can/should be used.  What do you think?

**Comment by carolili** [comment]
Hello, thank you for the replies and sorry for my late reply! The explanations in your responses are very informative 👍 

On the release front I personally find the following useful:
- The changelog, i.e. what @jsanda posted above, this gives an overview of whether something important changed or was fixed.
- When to expect a new release, i.e. when I could anticipate bugfixes or necessary features to arrive, so that I can plan when to test them.
- As I understand it this doesn't seem to be a production-ready project yet, at least for certain use cases. There also seems to be a large milestone coming up with Kubecon. So, the changelogs, release cycle, and roadmaps would help me understand when I can plan to test a version that has the features that are important to my use case.

I'm not so much looking to influence the future content unless there is something blocking me from using the chart 😉 At the moment I'm still trying it out to evaluate if we can use it in production in the future. We have a fairly small use case and the important parts we find in K8ssandra is the inclusion of tools like Reaper and Medusa, i.e. having a complete solution for deploying Cassandra with and easy-to-use backups and repair functionality.

Thank you for your efforts!
